pkg/proxy/controller.go

```diff
@@ -126,19 +148,58 @@ func (c *Controller) ListenAndServe() {
 		logrus.Fatal("No valid TLS configuration found, please use -certfile/-keyfile, -autocert or -selfcert options")
 	}
 
-	listener, err := tls.Listen(c.Network, c.Address, &tlsConfig)
-	if err != nil {
-		logrus.Fatal(err)
-	}
-	defer listener.Close()
-	close(c.startchan) // Controller is listening.
-	logrus.Infof("Listening on %s", c.Address)
-	for {
-		conn, err := listener.Accept()
+	if strings.Contains(c.Address, "https://") {
+		//websocket listen
+		listener, err := tls.Listen(c.Network, strings.Replace(c.Address, "https://", "", 1), &tlsConfig)
```

Comment:
Maybe use the `net/url` package instead of strings.Replace : https://gobyexample.com/url-parsing

Reply:
I think - no difference..